lib/vrl/parser/src/parser.lalrpop

```diff
@@ -445,9 +445,13 @@ ElseIf: IfStatement =
     IfStatement { predicate, consequent, alternative: None }
 };
 
+StatementSeparator: () = {
+    NonterminalNewline, ";"
+};
+
 Predicate: Predicate = {
     Box<ArithmeticExpr> => Predicate::One(<>),
-    "(" <v:(<AssignmentExpr> ";")+> <e:(<AssignmentExpr>)?> ")" => {
+    "(" StatementSeparator? <v:(<AssignmentExpr> StatementSeparator)+> <e:(<AssignmentExpr>)?> ")" => {
```

Comment:
The first `StatementSeparator?` should be `NonterminalNewline*`

Also, I don't believe this makes it possible to have multiple newlines between expressions, which would be required if someone wanted to write:

```
if (
	# here's my foo check
	foo

	# and here's the bar check
	bar
) { ... }
```